lib/pure/asyncdispatch.nim

```diff
@@ -1234,7 +1234,7 @@ else:
           processBasicCallbacks(fd, writeList)
           result = true
 
-        if Event.User in events or events == {Event.Error}:
+        if Event.User in events:
```

Comment:
Could we keep this and instead change:

```
            if rLength == 0: 
```

To

```
            if rLength == 0 and Event.User in events: 
```

My concern is that selectors containing only "User" events will silently fail.

Reply:
For keys which has only `{Event.Error}` we schedule callbacks 2 times... This is mistake.
For issue sourcecode it sometimes return `{Event.Read, Event.Error}` but sometimes it only gets `{Event.Error}`, and this is not bug of `ioselectors.nim`, epoll returns `EPOLLIN|EPOLLHUP` in some cases and in some cases it only returns `EPOLLHUP` for selected fd.

We already processed this callback here:
```nim
        if Event.Read in events or events == {Event.Error}:
          processBasicCallbacks(fd, readList)
          result = true
```

And we must not schedule it second time here, and you trying to apply patch after callback schedule.

Reply:
What if only custom user events are registered with nothing else? The following code:

```nim
        if Event.Read in events or events == {Event.Error}:
          processBasicCallbacks(fd, readList)
          result = true
```

Won't be executed. No?

Reply:
I don't care about custom user events, because i think it must be deprecated or not be part of `asyncdispatch`. In most cases this is just simple pipes and make them different from {Event.Read} was my mistake.

And answer on your question - This code would not be executed because custom user events got market with {Event.User} not {Event.Read}, but if some rare error could happen, then this code will be executed but not real {Event.User} code.

Reply:
Alright, I still have some reservations but I'm gonna merge this nonetheless. Thanks for answering my questions :)